src/Miningcore/Persistence/Postgres/Repositories/BalanceRepository.cs

```diff
@@ -70,6 +75,12 @@ public async Task<int> AddAmountAsync(IDbConnection con, IDbTransaction tx, stri
 
             if(balance == null)
             {
+                if(deduction > 0 && deduction < amount)
```

Comment:
We should keep biz logic away from the data layer as it is hard to maintain.

Reply:
I had the same thought,  but this function already has some business logic, and the alternative would be do break the addAmount function up into separate calls to getbalance and addBalance so we can access the old balance at the business layer.  you think that's the right solution?

Reply:
Yeah, that seems much better but no need to alter IPayoutScheme since it's very specific.